packages/common/src/directives/ng_optimized_image/url.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// Converts a string that represents a URL into a URL class instance.
+export function getUrl(src: string, win: Window): URL {
+  // Don't use a base URL is the URL is absolute.
+  return isAbsoluteUrl(src) ? new URL(src) : new URL(src, win.location.href);
+}
+
+// Checks whether a URL is absolute (i.e. starts with `http://` or `https://`).
+export function isAbsoluteUrl(src: string): boolean {
+  return /^https?:\/\//.test(src);
```

Comment:
`//example.com/...` is a valid URL too.

Reply:
so could be like this?

```suggestion
  return /^(?:https?:)?\/\//.test(src);
```

Reply:
The `?:` part is not necessary as the group is not used anyway, but yes. This should do the trick.

Reply:
Thanks for the feedback, we'll review this and create a followup PR if needed.

Reply:
I'm not sure if we should we encourage protocol-relative URLs given https://www.paulirish.com/2010/the-protocol-relative-url/ 

> Now that SSL is [encouraged for everyone](https://www.eff.org/encrypt-the-web-report) and [doesn’t have performance concerns](https://istlsfastyet.com/), this technique is now an anti-pattern. If the asset you need is available on SSL, then always use the https:// asset.
Allowing the snippet to request over HTTP opens the door for attacks like the [recent Github Man-on-the-side attack](http://www.netresec.com/?page=Blog&month=2015-03&post=China%27s-Man-on-the-Side-Attack-on-GitHub). It’s always safe to request HTTPS assets even if your site is on HTTP, however the reverse [is not true](https://developer.mozilla.org/en-US/docs/Security/MixedContent/How_to_fix_website_with_mixed_content).